cns/service/main.go

```diff
@@ -1234,17 +1234,14 @@ func InitializeCRDState(ctx context.Context, httpRestService cns.HTTPService, cn
 	switch {
 	case cnsconfig.ManageEndpointState:
 		logger.Printf("Initializing from self managed endpoint store")
-		podInfoByIPProvider, err = cnireconciler.NewCNSPodInfoProvider(httpRestServiceImplementation.EndpointStateStore) // get reference to endpoint state store from rest server
+		podInfoByIPProvider, err = InitializeStateFromCNS(cnsconfig, httpRestServiceImplementation.EndpointStateStore)
```

Comment:
instead of it being the responsibility of the restore state block to _also_ do the state migration, we should conditionally do the migration first. 
all of this stays unchanged, but in a block above, we can:
```go
if cnsconfig.EnableStateMigration {
    // if CNS statefile exists then:
        // short circuit, no need to do anything
    // else then:
        // construct CNI PodInfoProvider
        // convert PodInfoByIP to state format
        // write CNS statefile
}
// proceed to reconcile from CNS state as normal
```

Reply:
this is a major blocker for me

Reply:
Addressed in the latest commit.